### PR TITLE
Neo4j auth endpoints

### PR DESCRIPTION
**Pull request by Moshizzl3**
- Added auth endpoints using neo4j instead
- Added small change to client so we can switch between endpoints to demo our implementation

### CODE REVIEW

File: rpg_api/web/api/neo4j/auth/auth_dependencies.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    dao = NeoBaseUserDAO(session=session)
+    try:
+        return await dao.get_by_id(
+            id=token_data.user_id,  # type: ignore
+        )
+    except exceptions.RowNotFoundError:
+        raise exceptions.HttpNotFound("Decoded user not found.")
+
+
+# async def get_character_if_user_owns(
```
Comment (Moshizzl3) [inline rpg_api/web/api/neo4j/auth/auth_dependencies.py:50]: will work on this in next pr.

**Review by martinlaursen97 — APPROVED** [review]
you can keep it if you think it's a waste of time, but i have just reused the auth util stuff from postgres  in my PR, since we need to do the exact same thing for every db implementation

File: rpg_api/web/api/neo4j/auth/token_store.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    def get(self, user_id: int) -> str | None:
+        """Get the token."""
+        return self.tokens.get(user_id)
+
+    def pop(self, user_id: int) -> str | None:
+        """Pop the token."""
+        return self.tokens.pop(user_id, None)
+
+
+token_store = InMemoryTokenStore()
```
[the remark this answers is not in the document]
Comment (Moshizzl3) [reply rpg_api/web/api/neo4j/auth/token_store.py:21]: I need this one, user_id is an int

File: rpg_api/web/api/neo4j/auth/auth_dependencies.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    token: str = Depends(get_token),
+) -> dtos.BaseUserDTO:
+    """Get current user from token data."""
+    token_data = auth_utils.decode_token(token)
+    dao = NeoBaseUserDAO(session=session)
+    try:
+        return await dao.get_by_id(
+            id=token_data.user_id,  # type: ignore
+        )
+    except exceptions.RowNotFoundError:
```
[the remark this answers is not in the document]
Comment (Moshizzl3) [reply rpg_api/web/api/neo4j/auth/auth_dependencies.py]: Yea I agree, i have cleaned it up, but now the import happens from here `from rpg_api.web.api.postgres.auth.auth_dependencies import get_token`

I think we should move the auth_dependencies out, so they dont "belong" to postgres.

File: rpg_api/web/api/neo4j/auth/token_store.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (martinlaursen97) [reply rpg_api/web/api/neo4j/auth/token_store.py:21]: we could just let it use either an int or uuid
```python
from uuid import UUID

Key = UUID | int


class InMemoryTokenStore:
    """In memory token store (Cheap redis clone)."""

    def __init__(self) -> None:
        self.tokens: dict[Key, str] = {}

    def set(self, key: Key, token: str) -> None:
        """Set the token."""
        self.tokens[key] = token

    def get(self, key: Key) -> str | None:
        """Get the token."""
        return self.tokens.get(key)

    def pop(self, key: Key) -> str | None:
        """Pop the token."""
        return self.tokens.pop(key, None)


token_store = InMemoryTokenStore()

```

**Pull request merged**